elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchProject.java

```diff
@@ -68,6 +68,10 @@
       final String name = pair.right;
```

Comment:
I think I understand now why you are obliged to put explicit aliases in the query. It is because you are relying in the method getNamedProjects() just above. I think it would be better if you just used the method getProjects() and then identify which RexNodes correspond to Elastic fields using a RexVisitor similar to the one use in ElasticSearchRules.isItem(RexNode).